packages/modules/payment/src/services/payment-module.ts

```diff
@@ -434,14 +436,19 @@ export default class PaymentModuleService<
       return await this.baseRepository_.serialize(payment, { populate: true })
     }
 
-    const { data, status } =
-      await this.paymentProviderService_.authorizePayment(
-        {
-          provider_id: session.provider_id,
-          data: session.data,
-        },
-        context
-      )
+    let { data, status } = await this.paymentProviderService_.authorizePayment(
```

Comment:
q: should we try/catch here and throw a generic medusa error?

Reply:
we already handle that in the payment provider service

Reply:
That's a good point, there is a `PaymentSessionStatus.ERROR` as a type but at the moment the only provider we have is never returning that.
We have to refactor the Payment module to not block the db transaction when calling the provider, and I will test this entire flow a bit more in a next PR.

Reply:
but if you look at what we have done in the providers, we return some errors then the provider-service get that error and throw a MedusaError, so you should be good in that regards

Reply:
you mean in v1? because in v2, I don't see that.

Reply:
Yes in v2 as well, if the provider sdk throw we catch and return a formatted error and in the provider service then this.throwPaymentProviderError(error), otherwise we throw a medusa error from the provider if it is un handled. You can look at the payment/src/services/provider service

Reply:
I see, I was looking at the stripe implementation where authorizePayment isn't handling anything because it is just reading the status.